### PR TITLE
Fixed casting warnings from strtoul on macOS, change port to uint16_t.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: util.c [hunk 1]
```diff
@@ -437,7 +437,7 @@ char *convert_proxy_list_to_uri_list(const char *proxy_list, const char *default
         // Determine scheme for type PROXY
         if (!scheme) {
             // Parse port from host
-            int32_t port = get_host_port(host_start, host_len, 80);
+            uint16_t port = get_host_port(host_start, host_len, 80);
```
Comment (sergio-nsk) [inline util.c:440]:
```suggestion
            const uint16_t port = get_host_port(host_start, host_len, 80);
```

Comment (nmoinvaz) [reply util.c]: Fixed.

File: util.c [hunk 2]
```diff
@@ -485,7 +485,7 @@ bool should_bypass_proxy(const char *url, const char *bypass_list) {
         return true;
 
     // Strip and parse port from host
-    int32_t host_port = strip_host_port(host, strlen(host), 0);
+    uint16_t host_port = strip_host_port(host, strlen(host), 0);
```
Comment (sergio-nsk) [inline util.c:488]: Can you move it closer to the usage?

Comment (nmoinvaz) [reply util.c:488]: I prefer to keep it with the other host parsing code.

**Review by sergio-nsk — APPROVED** [review]

**Pull request merged**